common/components/common/owners/ChangeOwnerButton.jsx

```diff
@@ -0,0 +1,82 @@
+// @flow
+
+import React from "react";
+import Button from "react-bootstrap/Button";
+import type { ProjectDetailsAPIData } from "../../utils/ProjectAPIUtils.js";
+import type { EventData } from "../../utils/EventAPIUtils.js";
+import type { GroupDetailsAPIData } from "../../utils/GroupAPIUtils.js";
+import CurrentUser from "../../utils/CurrentUser.js";
+import metrics from "../../utils/metrics.js";
+import ChangeOwnerModal from "./ChangeOwnerModal.jsx";
+import _ from "lodash";
+
+type Props = {|
+  project: ?ProjectDetailsAPIData,
+  event: ?EventData,
+  group: ?GroupDetailsAPIData,
+|};
+type State = {|
+  showModal: boolean,
+  buttonDisabled: boolean,
+  buttonTitle: string,
+|};
+
+/**
+ * Button to open Modal for changing owner of project.
+ */
+class ChangeOwnerButton extends React.PureComponent<Props, State> {
+  constructor(props: Props): void {
+    super(props);
+
+    this.state = {
+      showModal: false,
+    };
+    this.handleShow = this.handleShow.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  handleShow() {
+    /*metrics.logGroupInviteProjectClick(
+      CurrentUser.userID(),
+      this.props.project.project_id
+    );*/
+    this.setState({ showModal: true });
+  }
+
+  closeModal() {
+    this.setState({ showModal: false });
+  }
+
+  displayChangeOwnerButton(): ?React$Node {
+    return (
+      <div>
+        <Button
+          variant="primary"
+          className={(this.props.project && "AboutProject-button") || this.props.event && "AboutEvent-edit-btn" || "AboutGroup-button"}
```

Comment:
@PeterBreen I would love you to take a look at this. I think it's really ugly. Also the Event css class lands up hiding the existing Edit Event button.

Reply:
What are the steps I should take to get this to render so I can look at it in the browser? I'm in my test environment viewing both projects and groups which my account owns; not seeing it show. Are there some conditions I need to meet aside from being logged in as project owner, viewing a group/project I own, and having fellow volunteers on the team?

I've got some ideas on how to proceed based on viewing the CSS rules and guessing but that's not really the best idea  for going forward 🙃

Reply:
I'm sorry - I didn't see your comment before.

This button only appears for Staff - so for me, it works with the Test Admin user.

As I mentioned in the meeting, I did have a bit of strange behavior where the correct user was logged in but it wasn't showing. That resolved by itself and has been stable ever since.